main.py

```diff
@@ -7,36 +7,42 @@
 from apscheduler.schedulers.background import BackgroundScheduler
 from pytz import utc
 
-from usecase import Lunchbreak, WorkSession
-
-usecaseByContext = {
-    "lunch": Lunchbreak,
-    "work": WorkSession
-}
+from usecase import Lunchbreak
+from handler import UsecaseStore
 
 hostName = "localhost"
 serverPort = 9150
 
-def schedule_usecases(scheduler):
-    scheduler.add_job(func=tick, trigger='interval', seconds=3)
-    scheduler.add_job(func=Lunchbreak().trigger_proactive_usecase,
-                      args=(),
-                      trigger='interval',
-                      hours=1)
+if __name__ == '__main__':
 
-def tick():
-    print('Tick! The time is: %s' % datetime.now())
+    def block_trigger(usecase, func, *args, **kwargs):
```

Comment:
Triggers are wrapped by this such that nothing proactive will start running while other usecases are running. If the usecase that is running is the proactive one, ignore the trigger, else register it as a callback on finish.